Updated CardClient View

giiks131 opened the pull request:
- small UI Updates based on modifiers and paddings
- added action for card, that opening a test modal view

giiks131 committed:
added HomeView with ClientCard

giiks131 committed:
added plus button

giiks131 committed:
deployment minimum version

giiks131 committed:
deleted xuserdata

inTheAM committed:
Removed development team

giiks131 committed:
ClientCard UI updates

Update ClientCard UI with spacer and modifiers fixes, new image Asset

giiks131 committed:
ClientCard - added action for Card

Added test ModalView, that will show a client info by clicking on ClientCard

giiks131 committed:
Add MARK info for image

giiks131 commented:
<img width="349" alt="image" src="https://user-images.githubusercontent.com/82652206/211862193-d06c9d75-3c64-4627-931e-5272aaaa2bc2.png">

inTheAM commented on SlimApp/ClientCard.swift:
```diff
@@ -0,0 +1,59 @@
+                    .scaledToFit()
+            }
+            .padding(20)
+            .frame(maxWidth: .infinity, maxHeight: 110)
```
since you're using a Spacer(), wouldn't it be better to remove the infinity width? maybe just use a frame height

giiks131 replied:
agree, fixed it

inTheAM commented on SlimApp/ClientCard.swift:
```diff
@@ -0,0 +1,59 @@
+            .frame(maxWidth: .infinity, maxHeight: 110)
+            .background(.purple)
+            .foregroundColor(.white)
+            .mask(RoundedRectangle(cornerRadius: 28, style: .continuous))
```
Is this mask being used to set the corner radius of the card?

giiks131 replied:
you are absolutely right - this mask creates a nice-to-see UI view with editable corner

inTheAM replied:
In that case, consider applying a .cornerRadius to the purple color in the background.

giiks131 replied:
done

inTheAM commented on SlimApp/ClientCard.swift:
```diff
@@ -0,0 +1,59 @@
+            .mask(RoundedRectangle(cornerRadius: 28, style: .continuous))
+            .sheet(isPresented: $showingSheet) {
+                // MARK: Test ModalView, that will show a client info by clicking on ClientCard
+                TestModal()
```
Since the modal is being created by @yukim17, maybe you should leave the sheet empty. This TestModal is probably going to end up being deleted. So leaving it empty might be the better option and it can be filled with the modal later

giiks131 replied:
i think that it it useful to see the behavior that we except from tapping this button, but if you suggest to remove this View and just have a blank space — I'm ok with that

inTheAM commented on SlimApp/HomeView.swift:
```diff
@@ -0,0 +1,61 @@
+//                .padding(.horizontal, 20)
+
+
+//            Divider()
```
Is this code still going to be used or can it be discarded?

giiks131 replied:
fixed on another branch with ClientList

inTheAM commented on SlimApp/HomeView.swift:
```diff
@@ -0,0 +1,61 @@
+        ZStack {
+            Color("Background")
+                .ignoresSafeArea()
+            NavigationView {
```
Maybe consider fixing this ZStack/NavigationView order

giiks131 replied:
fixed on another branch with ClientList

inTheAM committed:
Delete .gitignore

giiks131 committed:
Fixes after Review

Clean code for HomeView and fix background issue

giiks131 committed:
Fixes after Review

Clean code and deleted TestModal()

giiks131 committed:
Fixed after Review // corner radius

Put cornerRadius instead of mask

inTheAM commented on SlimApp/ClientCard.swift:
```diff
@@ -0,0 +1,50 @@
+            .cornerRadius(28)
+            .foregroundColor(.white)
+            .sheet(isPresented: $isShowingSheet) {
+                // MARK: open ModalView, that will show a client info by clicking on ClientCard
```
I think the modal view is supposed to be in the main screen, to be used to input client details.
This client card should use navigation to push the screen showing client details.
Consider changing this button to a navigation link that pushes an empty view

giiks131 replied:
done after discussion

giiks131 committed:
Fix after Review // Opening new Views

Delete opening new modal view from ClientCard. Add NavigationLink for ClientCard inside ForEach() in HomeView(). Add possibility to create a blank Modal sheet for toolBar button

giiks131 committed:
Fixes after Review // List of clients

Remove extension with list of clients card and leave only logic for HomeView

giiks131 committed:
Merge remote-tracking branch 'refs/remotes/origin/main'

Conflicts:
	SlimApp.xcodeproj/project.pbxproj

giiks131 committed:
delete MARK

giiks131 committed:
delete contentView
